pupynere.py

```diff
@@ -1056,15 +1070,20 @@ def __setitem__(self, index, data):
                 recs = rec_index + 1
             if recs > len(self.data):
                 shape = (recs,) + self._shape[1:]
-                self.data = np.resize(self.data, shape)
-        self.data[index] = data
+                self.__dict__["data"] = np.resize(self.data, shape)
+        self.__dict__["data"][index] = data
     
     def _data_allocated(self):
-        return self.data is not None
+        return hasattr(self, "data") and self.data is not None
```

Comment:
Shouldn't `hasattr(self, "data")` *always* be `True` since it gets set in `__init__()`?